### PR TITLE
Create store

### PR DESCRIPTION
**Pull request by liambarstad**
#### What does  this PR do?
+ allows registered users from the main page to create a pending store, which they can then access and see the stores they have created, they are initialized as stores admin
+ the schema posted in slack has been implemented, and should not cause any difficulties
+ a StoreService object (factory) has been created, with a method .create_proposed_store(user, store_name) which instantiates a pending store and makes the user the admin of that store
+ tests for StoreService and the user story
#### What are the relevant story numbers?
#153568630
#### Questions:
  - Do Migrations Need to be ran? -> Yes
  - Do Environment Variables need to be set? -> No
  - Any other deploy steps? -> No

PS: with this implementation, the admin dashboard is not yet available to registered users who are store admins

#### Pivotal hook 
- [(Finishes|Fixes|Delivers) #153568630]

### CODE REVIEW

File: app/models/role.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Role < ApplicationRecord
+end
```
Comment (dionew1) [inline app/models/role.rb]: Could you add the association to users, please?

File: app/views/dashboard/index.html.erb [hunk 2]
```diff
@@ -8,6 +8,8 @@
         <li class="nav-item">
           <a class="nav-link" href="/orders">My Orders</a>
         </li>
+        <li class="nav-item">
+          <%= link_to "My Stores", users_stores_path, class: "nav-link" %>
```
Comment (dionew1) [inline app/views/dashboard/index.html.erb:12]: missing closing tag for li

File: app/views/users/stores/_store.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<tr id="store-<%= store.id %>" class="store">
+  <th scope="row">Insert Image Here</th>
```
Comment (dionew1) [inline app/views/users/stores/_store.html.erb]: Is this an image for the store? 
It's unclear to me what this is for.

File: spec/models/role_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Role, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (dionew1) [inline spec/models/role_spec.rb]: Can you add some shoulda validations for relationships here?

File: spec/models/store_spec.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Store, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (dionew1) [inline spec/models/store_spec.rb]: Can you add some shoulda validations for relationships here?

File: spec/models/user_role_store_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe UserRoleStore, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (dionew1) [inline spec/models/user_role_store_spec.rb]: Can you add some shoulda validations for relationships here?

File: app/views/dashboard/index.html.erb (hunk 2 above)
Comment (liambarstad) [reply app/views/dashboard/index.html.erb:12]: thank you! almost missed it

File: app/views/users/stores/_store.html.erb (hunk 3 above)
Comment (liambarstad) [reply app/views/users/stores/_store.html.erb]: pretty much, i'm assuming that when we have to create a stores#index and show page, we will need images, I'll take that line out for now

File: spec/factories/user_role_stores.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :user_role_store do
+    user nil
+    role nil
+    store nil
```
Comment (hmorri32) [inline spec/factories/user_role_stores.rb]: Build factories || don't stage if not built

File: spec/models/role_spec.rb (hunk 4 above)
Comment (hmorri32) [reply spec/models/role_spec.rb]: write tests || don't stage if no tests

**Pull request merged**